winsup/utils/getprocaddr.c

```diff
@@ -40,12 +40,23 @@ inject_remote_thread_into_process (HANDLE process,
     {
       /*
        * Wait up to 10 seconds (arbitrary constant) for the thread to finish;
-       * After that grace period, fall back to exit with an exit code
-       * indicating failure.
+       * Maybe we should wait forever? I have seen Cmd does so, but well...
        */
-      if (WaitForSingleObject (thread, 10000) == WAIT_OBJECT_0 &&
-          GetExitCodeThread(thread, &code) && code != STILL_ACTIVE)
+      if (WaitForSingleObject (thread, 10000) == WAIT_OBJECT_0)
         res = 0;
+      /*
+      According to the docs at MSDN for GetExitCodeThread, it will
+      get the return value from the function, here CtrlRoutine. So, this
+      checks if the Ctrl Event is handled correctly by the process.
+
+      By some testing I could see CtrlRoutine returns 0 in case where
+      CtrlEvent set by SetConsoleCtrlHandler is handled correctly, in all
+      other cases it returns something non-zero(not sure what it that).
+      */
+      GetExitCodeThread (thread, &code);
+      if (code != 0)
+        res = code;
```

Comment:
It seems a little weird to conflate an error injecting the thread with the return value from the injected thread.  But I guess it works...